app/controllers/concerns/new_device_concern.rb

```diff
@@ -1,8 +1,12 @@
 # frozen_string_literal: true
 
 module NewDeviceConcern
-  def set_new_device_session
-    user_session[:new_device] = !current_user.authenticated_device?(cookie_uuid: cookies[:device])
+  def set_new_device_session(new_device = nil)
```

Comment:
since it looks like there are only 2 callsites for this method, what if we made it a required param instead of an optional one?

Reply:
> since it looks like there are only 2 callsites for this method, what if we made it a required param instead of an optional one?

Do you mean updating the callsites to call it as `set_new_device_session(nil)` or as `set_new_device_session(!current_user.authenticated_device?(cookie_uuid: cookies[:device]))` ?

Reply:
Updated to be non-optional in 29b7673f2.